tests/test_unit/test_prerun_transformations/test_transformations.py

```diff
@@ -0,0 +1,633 @@
+    ),
+    # AddFromOffset test cases
+    TransformationTestCase(
+        case_id="AddFromOffset: case_1",
+        transformation=AddFromOffset(
+            relative_datapoints=[
+                RelativeDatapoint(offset_minute=0, offset_value=1),
+                RelativeDatapoint(offset_minute=20, offset_value=-2),
+                RelativeDatapoint(offset_minute=230, offset_value=3),
+            ]
+        ),
+        input_values=[42.0] * 6,
+        expected_values=[43, 40, 40, 40, 40, 45, 45, 45],
+        expected_times=[
+            datetime(2022, 1, 1, 0, tzinfo=None),
+            datetime(2022, 1, 1, 0, 20, tzinfo=None),
+            datetime(2022, 1, 1, 1, tzinfo=None),
+            datetime(2022, 1, 1, 2, tzinfo=None),
+            datetime(2022, 1, 1, 3, tzinfo=None),
+            datetime(2022, 1, 1, 3, 50, tzinfo=None),
+            datetime(2022, 1, 1, 4, tzinfo=None),
+            datetime(2022, 1, 1, 5, tzinfo=None),
+        ],
+        frequency="h",
+    ),
+]
+
+
+@pytest.mark.parametrize(
+    "test_case",
+    [pytest.param(test_case, id=test_case.case_id) for test_case in TRANSFORMATION_TEST_CASES],
+)
+def test_basic_transformations(test_case: TransformationTestCase):
+    """
+    Test for testing different transformations (AddConstant, Round, MultiplyConstant, ToBool, ToInt,
+        ZeroIfNotOne, OneIfTwo, DoNothing, MultiplyFromOffset, AddFromOffset)
+    """
+
+    incremental_dates = pd.date_range(
+        start=test_case.start_date, periods=len(test_case.input_values), freq=test_case.frequency
+    )
+
+    input_data = pd.Series(test_case.input_values, index=incremental_dates)
+
+    expected_data_index = incremental_dates if not test_case.expected_times else test_case.expected_times
+    expected_data = pd.Series(test_case.expected_values, index=expected_data_index)
+
+    output_data = test_case.transformation.apply(time_series_data=(input_data,))
+
+    pd.testing.assert_series_equal(expected_data, output_data, check_dtype=False, check_freq=False)
+
+
+def test_sum_timeseries():
+
+    input_values_1 = [42.0] * 5
+    input_values_2 = [20.0] * 5
+    input_values_3 = [15] * 10
+
+    start_time = "2022-01-01"
+    incremental_dates_1 = pd.date_range(start=start_time, periods=len(input_values_1), freq="h")
+    incremental_dates_2 = pd.date_range(start=start_time, periods=len(input_values_2), freq="2h")
+    incremental_dates_3 = pd.date_range(start=start_time, periods=len(input_values_3), freq="30min")
+
+    input_data_1 = pd.Series(input_values_1, index=incremental_dates_1)
+    input_data_2 = pd.Series(input_values_2, index=incremental_dates_2)
+    input_data_3 = pd.Series(input_values_3, index=incremental_dates_3)
+
+    input_dates = [pd.Series(incremental_dates_1), pd.Series(incremental_dates_2), pd.Series(incremental_dates_3)]
+    input_data = (input_data_1, input_data_2, input_data_3)
+
+    expected_data_1 = input_values_1
+    expected_data_2 = [62, 42, 62, 42, 62, 20, 20]
```

Comment:
I would expect [62, 62, 62, 62, 62, 62, 62] if we are summing 42 and 20 (if we assume is_step=true, which I think we should, one value is valid until the next value)